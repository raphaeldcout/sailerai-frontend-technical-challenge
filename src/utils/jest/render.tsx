import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

