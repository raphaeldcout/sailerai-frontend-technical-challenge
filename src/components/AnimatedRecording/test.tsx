import '@testing-library/jest-dom';

import { render } from '@/utils/jest/render';
import { screen } from '@testing-library/react';

import { AnimatedRecording } from '.';

describe('AnimatedRecording', () => {
  test('should match snapshot', () => {
    const { container } = render(<AnimatedRecording />);

    expect(container).toMatchSnapshot();
  });

  test('should show (N) wave from quantity props', () => {
    const n = 10;
    
    render(<AnimatedRecording quantity={n} />);

    const waveElements = screen.queryAllByTestId('root-recording-wave');

    expect(waveElements.length).toBe(n);
  });
});
