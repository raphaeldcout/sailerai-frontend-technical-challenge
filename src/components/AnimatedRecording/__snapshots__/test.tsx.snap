// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedRecording should match snapshot 1`] = `
<div>
  <div
    class="sc-6bc181fa-0 kpKwOb"
  >
    <div
      class="sc-6bc181fa-1 dsCirH"
      color="surface"
      data-testid="root-recording-wave"
      quantity="5"
    />
    <div
      class="sc-6bc181fa-1 dsCirH"
      color="surface"
      data-testid="root-recording-wave"
      quantity="5"
    />
    <div
      class="sc-6bc181fa-1 dsCirH"
      color="surface"
      data-testid="root-recording-wave"
      quantity="5"
    />
    <div
      class="sc-6bc181fa-1 dsCirH"
      color="surface"
      data-testid="root-recording-wave"
      quantity="5"
    />
    <div
      class="sc-6bc181fa-1 dsCirH"
      color="surface"
      data-testid="root-recording-wave"
      quantity="5"
    />
  </div>
</div>
`;
